tools/cdk-build-tools/lib/feature-flag.ts

```diff
@@ -0,0 +1,69 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as semver from 'semver';
+
+type Flags = {[key: string]: any} | undefined
+
+/* eslint-disable jest/no-export */
+
+/**
+ * jest helper function to be used when testing feature flags.
+ *
+ * This function is for writing jest tests that test feature flags that are present in CDKv1
+ * and will be removed in CDKv2. This helps execute tests written for CDKv1 in the context of CDKv2.
+ *
+ * The version of CDK is determined by running `scripts/resolve-version.js`, and the logic is as follows:
+ *
+ * When run in CDKv1, the tests are executed as normal.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is set
+ * (via CDK app context) are executed but without setting the app context. This is because CDKv2
+ * defaults to the behaviour for the flag being enabled.
+ *
+ * When run in CDKv2, tests that are intended to verify the behaviour when the feature flag is not
+ * set are skipped.
+ */
+export function withFeatureFlag<T>(
```

Comment:
Maybe name this something like `withFutureBehavior()` or something.

I'd like the name to make very clear that it's a non-revertible flag.

Reply:
`withLegacyBehavior()` would replace both `withFeatureFlagDisabled()` and `withFeatureFlag(undefined)` ?

Reply:
| Method | Version | Behavior |
|------|-------|-------|
| withFutureBehavior | v1 | Set feature flag |
| withFutureBehavior | v2 | No context |
| withLegacyBehavior | v1 | No context (potentially add an additional test for featureFlag=false??) |
| withLegacyBehavior  | v2 | Skip |

Reply:
We could be running 2 tests for `withLegacyBehavior`, to make very very sure that no context == flag set to `false`.

Might be worth it, might not.

Reply:
Let's see what the difference is between my proposed naming and yours, because I'm confusing myself now on seeing whether or not those two variants are actually different. I have a feeling they are, but let's confirm :)

| Method | version | flags | Behavior |
|-------|-------|-------|----------|
| withFeatureFlag | v1 | - | Run test with flags (either present or absent) |
| withFeatureFlag | v2 | yes | No context |
| withFeatureFlag | v2 | no | Skip |
| withFeatureFlagDisabled | v1 | (n/a) | No context |
| withFeatureFlagDisabled | v2 | (n/a) | Skip |